07.ls_object/lib/screen.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require_relative 'normal_formatter'
+require_relative 'detail_formatter'
+
+class Screen
+  def initialize(stats_in_dir, options = {})
+    @stats_in_dir = stats_in_dir
+
+    @long_format = options[:long_format]
+    @reverse = options[:reverse]
+    @header = options[:header]
+  end
+
+  def show
+    output_blocks = []
+
+    output_blocks << create_output_block_with_file_stats
+    output_blocks << create_output_blocks_with_dir_stats
+
+    puts output_blocks.reject(&:empty?).join("\n\n")
+  end
+
+  private
+
+  def create_output_block_with_file_stats
+    @long_format ? DetailFormatter.new(@stats_in_dir['']).write : NormalFormatter.new(@stats_in_dir['']).write
+  end
```

Comment:
```suggestion
  def create_output_block_with_file_stats
    formatter_factory.new(@stats_in_dir['']).write
  end
  
  def formatter_factory
    @long_format ? DetailFormatter : NormalFormatter
  end
```

ファクトリメソッドを使うとよりスッキリするかも
（これもデザインパターンの一種ですね）